google/cloud/storage/bucket.py

```diff
@@ -2361,7 +2361,13 @@ def lifecycle_rules(self):
             elif action_type == "SetStorageClass":
                 yield LifecycleRuleSetStorageClass.from_api_repr(rule)
             else:
-                raise ValueError("Unknown lifecycle rule: {}".format(rule))
+                warnings.warn(
+                    "Unknown lifecycle rule by the client: {}. Please upgrade your client.".format(
```

Comment:
I would say something like Unknown lifecycle rule type received: {}. Please upgrade to the latest version of google-cloud-storage.

How does that sound?

Reply:
That sounds much better. Thanks for the review!